internals/link_helpers.py

```diff
@@ -46,17 +55,62 @@ def __init__(self, url: str):
     self.is_error = False
     self.information = None
 
+  def _parse_github_markdown(self) -> dict[str, object]:
+    parsed_url = urlparse(self.url)
+    path = parsed_url.path
+    owner = path.split('/')[1]
+    repo = path.split('/')[2]
+    ref = path.split('/')[4]
+    file_path = '/'.join(path.split('/')[5:])
+    try:
+      # try to get the branch information, if it exists, update branch name
+      # this handles the case where the branch is renamed
+      branch_information = github_api_client.repos.get_branch(
+          owner=owner, repo=repo, branch=ref)
+      ref = branch_information.name
+    except HTTPError as e:
+      # if the branch does not exist, then it is probably a commit hash
+      if e.code != 404:
+        raise e
+
+    information = github_api_client.repos.get_content(
+        owner=owner, repo=repo, path=file_path, ref=ref)
+
+    # decode the content from base64
+    content_str = information.content
+    content_decoded = base64.b64decode(content_str).decode('utf-8')
+    # get markdown title, remove beginning and trailing # but keep the rest
+    title = content_decoded.split('\n')[0].strip('#').strip()
+    information['_parsed_title'] = title
+
+    return information
+
+  def _parse_github_issue(self) -> dict[str, object]:
+    """Parse the information from the github issue tracker."""
+
+    parsed_url = urlparse(self.url)
+    path = parsed_url.path
+    owner = path.split('/')[1]
+    repo = path.split('/')[2]
+    issue_id = path.split('/')[4]
+
+    information = github_api_client.issues.get(
```

Comment:
What errors do we need to handle?  What happens if this issue does not exist or the repo is not public?